### ISSUE TITLE
torch.max_pool and torch.nn.functional.max_pool should have matching signature (only nn namespace takes return_indices param)

### ISSUE THREAD

**Issue by heitorschueroff** [pytorch/pytorch#45851]
## 🐛 Bug

torch.max_pool... does not take return_indices param and instead implements a torch.max_pool_with_indices version while torch.nn.functional.max_pool... does take return_indices param.

## Expected behavior

The signatures should match.

## Additional context

Related issue https://github.com/pytorch/pytorch/issues/45494

**Comment by vadimkantorov** [comment]
Related: https://github.com/pytorch/pytorch/issues/44991

**Comment by volcacius** [comment]
dropout and alpha_dropout have this issue too:
`torch.dropout: lambda input, p, train, inplace=False: -1` vs `torch.nn.functional.dropout: lambda input, p=0.5, training=True, inplace=False: -1`
